### PR TITLE
Add ability to restart the app from the error screen #14

### LINKED ISSUE #14: Add ability to restart the app from the error screen
The crash screen should provide the user with the ability to restart the app.

Tasks:
- Record information to build an intent for the app when it crashes
- Add a button below _Share Stacktrace_ which fires the intent to launch the app again
- Finish the crashed activity

I am not sure how this would affect the task-stack. Might need to fiddle with launch options to fix unexpected behaviours.

### PR DESCRIPTION
**Pull request by shivam21**
What does this PR do?
* Add a button below Share Stacktrace which fires the intent to launch the app again.
WhatsApp Image 2020-10-03 at 11 48 53 AM

Fixes #14

### CODE REVIEW

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt [hunk 1]
```diff
@@ -31,11 +31,11 @@ class WhatTheStackActivity : AppCompatActivity() {
...
 
         window.decorView.systemUiVisibility =
-            View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
 
         Insetter.builder()
-            .applySystemWindowInsetsToPadding(Side.LEFT or Side.RIGHT or Side.TOP)
-            .applyToView(findViewById(R.id.nestedScrollRoot))
+                .applySystemWindowInsetsToPadding(Side.LEFT or Side.RIGHT or Side.TOP)
+                .applyToView(findViewById(R.id.nestedScrollRoot))
```
Comment (haroldadmin) [inline what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: Please refrain from modifying unrelated lines. Sometimes it's necessary, but not in this case.

File: what-the-stack/src/main/res/layout/activity_what_the_stack.xml [hunk 2]
```diff
@@ -95,7 +95,21 @@
...
-            app:iconTint="?attr/colorSecondary"/>
+            app:iconTint="?attr/colorSecondary" />
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/launchApplication"
+            style="@style/Widget.MaterialComponents.Button.OutlinedButton.Icon"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="2dp"
```
Comment (haroldadmin) [inline what-the-stack/src/main/res/layout/activity_what_the_stack.xml:106]: While the top margin on this button is correct, the bottom margin on the button above it is too high. Please change it to 2dp for consistent spacing in between all buttons.

**Review by haroldadmin — CHANGES_REQUESTED** [review]
Very nice work!

I have left some feedback on a couple of things that need to be changed. Once they are done, this PR will be ready to merge.

File: what-the-stack/src/main/res/layout/activity_what_the_stack.xml (hunk 2 above)
Comment (shivam21) [reply what-the-stack/src/main/res/layout/activity_what_the_stack.xml:106]: actually the above Share StackTrace button has marginBottom 8dp.
Should I change margin Bottom of Share StackTrace button to 2dp too?

Comment (haroldadmin) [reply what-the-stack/src/main/res/layout/activity_what_the_stack.xml:106]: Yes, that's exactly what I meant.

Comment (shivam21) [reply what-the-stack/src/main/res/layout/activity_what_the_stack.xml:106]: done

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt (hunk 1 above)
Comment (shivam21) [reply what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: done

**Review by haroldadmin — APPROVED** [review]
Looks good! Thanks for the quick turn-around on this.
